ci/deploy-vmck

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash -ex
+
+cd "$( dirname "${BASH_SOURCE[0]}" )"
+
+export VAGRANT_DOTFILE_PATH=$(mktemp -d --tmpdir VAGRANT_DOTFILE_XXXXXXXX)
+export VAGRANT_DEFAULT_PROVIDER=vmck
+export VAGRANT_CHECKPOINT_DISABLE=true
+export VAGRANT_BOX_UPDATE_CHECK_DISABLE=true
+
+TIMEOUT_MIN=23
+RETRIES=3
+
+function print_section() {
+  set +x
+  echo
+  echo '-----------------------------------------'
+  echo "| $1"
+  echo '-----------------------------------------'
+  set -x
+}
+
+function vagrant_up() {
+  set +e
+  vagrant up --no-provision || echo "vagrant up failed, VM might still work"
+  echo "sudo shutdown +$TIMEOUT_MIN" | vagrant ssh
+  sshret=$?
+  if [ 0 -eq $sshret ]; then
+    return 0
+  else
+    vagrant destroy -f
+    return 1
+  fi
+}
+
+function retry_vagrant_up() {
+  for i in $(seq 1 $RETRIES); do
+    print_section "Starting vagrant... (try #$i/$RETRIES)"
+    if vagrant_up; then
+      return 0
+    fi
+  done
+  echo "Vagrant failed after $RETRIES tries"
+  exit 1
+}
+
+retry_vagrant_up
+
+print_section "Run Script"
+set +e
+vagrant provision
+ret=$?
+
+
+print_section "Doing tests"
+vagrant ssh <<'EOF'
+set -e
+bash /opt/vmck/ci/test-pytest.sh
+bash /opt/vmck/ci/test-alive-on-cluster.sh
+EOF
+ret=$?
+
+print_section "Destroying Vagrant"
+vagrant destroy -f || echo "vagrant destroy failed, but we don't care"
+exit $ret
```

Comment:
See if we can get rid of all the uses of `$?`. If you really want to call `vagrant destroy` at the end, register an [exit trap](http://redsymbol.net/articles/bash-exit-traps/).